pkg/api/util/doget.go

```diff
@@ -39,22 +38,34 @@ func GetClient(verify bool) *http.Client {
 	return GetClientWithTimeout(0, verify)
 }
 
+// ipcRoundTripper is an implementation of http.RoundTripper interface
+type ipcRoundTripper struct {
+	tr http.Transport
+}
+
+// RoundTrip implement the http.RoundTripper interface method to be used in http request context
+// It is used to lazy load the TLS Client config, in order to let the time to initialize it before using it
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏This comment kind of hints that there is a race, is it the case ?

Reply:
This is due to historical choices: since the creation of authentication artifacts is done in parallel and their creation time is unpredictable, we delay the fetch as much as possible to increase the chances of having them initialized.
This should be reworked to achieve better consistency.